samtranslator/schema/schema.py

```diff
@@ -45,42 +45,84 @@ class AwsServerlessConnector(BaseModel):
     Properties: ConnectorProperties
 
 
+class FunctionCodeUri(BaseModel):
+    Bucket: Union[str, SamIntrinsic]
+    Key: Union[str, SamIntrinsic]
+    Version: Optional[Union[str, SamIntrinsic]]
+
+
+class FunctionDeploymentPreferenceHooks(BaseModel):
+    PostTraffic: Optional[Union[str, SamIntrinsic]]
+    PreTraffic: Optional[Union[str, SamIntrinsic]]
+
+
+class FunctionDeploymentPreference(BaseModel):
+    Alarms: Optional[Union[List[SamIntrinsic], SamIntrinsic]]
+    Enabled: Optional[Union[bool, SamIntrinsic]]
+    Hooks: Optional[FunctionDeploymentPreferenceHooks]
+    PassthroughCondition: Optional[Union[bool, SamIntrinsic]]
+    Role: Optional[Union[str, SamIntrinsic]]
+    TriggerConfigurations: Optional[PassThrough]
+    Type: Union[str, SamIntrinsic]
+
+
+class FunctionDeadLetterQueue(BaseModel):
+    TargetArn: str
+    Type: Literal["SNS", "SQS"]
```

Comment:
Let's separate these so the `Type` is coupled to whatever properties it has. Otherwise makes it a pain to expand later.

Reply:
Ignore, this one is fine. Still applies to `Events`.